docs/developer/architecture/configurations.rst

```diff
@@ -91,17 +91,45 @@ requirements for the following reasons:
 How it works
 ^^^^^^^^^^^^
 
-During the Bootstrap stages, when we are assertive that the K8s cluster is
-fully ready and available we could perform the following actions:
+During Bootstrap, Upgrade or Downgrade stages, when we are assertive that
+the K8s cluster is fully ready and available we could perform the following
+actions:
 
-  - Create and deploy ConfigMaps that hold cluster and service configurations
-    and pre-fill them with default values.
-  - Template service pods and deployments to consume configuration data
-    directly from the above deployed ConfigMaps
+  - Firstly, create and deploy ConfigMaps that will hold customizable cluster
+    and service configurations.
+    These ConfigMaps should define an empty `config.yaml` in the data section
+    of the ConfigMap for later use. A standard layout for each customizable
+    field could be added in the documentation to assist MetalK8s administrator
+    in adding and modifying customizations.
```

Comment:
One thing we could do to make this simpler: on Install/Upgrade/Downgrade, push the current default values in another key of this ConfigMap (say, `defaults.yaml`), and if possible (should be okay if we just needed to paste the contents of a single `csc.yaml` :wink:), with inline comments.

This makes the ConfigMaps self-documented. Now, I'm not saying we shouldn't document all of this, but when in a terminal to `kubectl edit` such a configuration, having everything at hand would be quite comfortable.

Of course, this key in the ConfigMap would only serve as documentation, and should never be used for deploying (in case someone edits it).